privacyidea/lib/policy.py

```diff
@@ -929,12 +942,100 @@ def filter_policies_by_conditions(self, policies, user_object=None, request_head
                 reduced_policies.append(policy)
         return reduced_policies
 
-    @staticmethod
-    def _policy_matches_request_environ_condition(policy, key, comparator, value, request_headers):
+    @classmethod
+    def _do_handle_missing_data(cls, handle_missing_data: str, policy_name: str, missing: str, section: str,
+                                object_name: str, key: str, available_keys: list = None) -> bool:
+        """
+        This function handles the behaviour of the system if the data that is required to check a policy condition is
+        missing. There are three valid options: raise an error (default), evaluate the condition to True or evaluate
+        the condition to False.
+
+        :param handle_missing_data: The name of the behaviour if the data is missing
+        :param policy_name: The name of the policy
+        :param missing: The missing data, either equal to the object_name or the key
+        :param section: The section of the condition
+        :param object_name: The name of the object that should be evaluated in the condition
+        :param key: The key that should be evaluated in th condition
+        :param available_keys: The available keys of the object (if it is available), optional
+        :return: True if handle_missing_data is CONDITION_IS_TRUE
+                 False if handle_missing_data is CONDITION_IS_FALSE
+                 raise PolicyError if handle_missing_data is RAISE_ERROR or is None
+        """
+        if not handle_missing_data or handle_missing_data == ConditionHandleMissingData.RAISE_ERROR:
```

Comment:
can this be refacored to have the check for `handle_missing_data` at the bottom? If i see it correctly it influences the log level and in one case it raises the error. So ideally the log texts would be there only once, so it looks less copy-pastey